stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -932,16 +934,17 @@ See also [`hermitianpart`](@ref).
 !!! compat "Julia 1.10"
     This function requires Julia 1.10 or later.
 """
-hermitianpart!(A::AbstractMatrix, uplo::Symbol=:U) =
-    Hermitian(_hermorsympart!(hermitianpart, adjoint, A, char_uplo(uplo)), uplo)
+hermitianpart!(A::AbstractMatrix, uplo::Symbol=:U, full::Bool = false) =
+    Hermitian(_hermorsympart!(a -> hermitianpart(a, uplo, full), adjoint, A, char_uplo(uplo), full), uplo)
 
-function _hermorsympart!(real::Function, conj::Function, A::AbstractMatrix, uplo::Char)
+@inline function _hermorsympart!(real::Function, conj::Function, A::AbstractMatrix, uplo::Char, full::Bool)
     require_one_based_indexing(A)
     n = checksquare(A)
     @inbounds for j in 1:n
         A[j, j] = real(A[j, j])
         for i in (uplo === 'U' ? (1:(j - 1)) : ((j + 1):n))
-            A[i, j] = (A[i, j] + conj(A[j, i])) / 2
+            A[i, j] = val = (A[i, j] + conj(A[j, i])) / 2
+            full && (A[j, i] = conj(val))
```

Comment:
Seems suboptimal to have this test inside the loop.   If it doesn't cost too much, why not just set `A[j, i]` unconditionally as @Jutho suggested?  I doubt that this function is performance-critical in any real application.